SmartDeviceLink/SDLSystemCapabilityManager.m

```diff
@@ -136,136 +128,26 @@ - (void)stop {
     _appServicesCapabilitiesDictionary = [NSMutableDictionary dictionary];
 
     _supportsSubscriptions = NO;
-    for (SDLSystemCapabilityType capabilityType in [self.class sdl_systemCapabilityTypes]) {
-        _capabilityObservers[capabilityType] = [NSMutableArray array];
-    }
+    [_capabilityObservers removeAllObjects];
+    [_subscriptionStatus removeAllObjects];
+
+    _currentHMILevel = SDLHMILevelNone;
 
-    _isFirstHMILevelFull = NO;
     _shouldConvertDeprecatedDisplayCapabilities = YES;
 }
 
 #pragma mark - Getters
 
+- (BOOL)supportsSubscriptions {
+    return [[SDLGlobals sharedGlobals].rpcVersion isGreaterThanOrEqualToVersion:[SDLVersion versionWithString:@"5.1.0"]];
+}
+
 - (nullable SDLAppServicesCapabilities *)appServicesCapabilities {
     if (self.appServicesCapabilitiesDictionary.count == 0) { return nil; }
 
     return [[SDLAppServicesCapabilities alloc] initWithAppServices:self.appServicesCapabilitiesDictionary.allValues];
 }
 
-#pragma mark - Notifications
-
-/**
- *  Registers for notifications and responses from Core
- */
-- (void)sdl_registerForNotifications {
-    [[NSNotificationCenter defaultCenter] addObserver:self selector:@selector(sdl_registerResponse:) name:SDLDidReceiveRegisterAppInterfaceResponse object:nil];
-    [[NSNotificationCenter defaultCenter] addObserver:self selector:@selector(sdl_displayLayoutResponse:) name:SDLDidReceiveSetDisplayLayoutResponse object:nil];
-    [[NSNotificationCenter defaultCenter] addObserver:self selector:@selector(sdl_systemCapabilityUpdatedNotification:) name:SDLDidReceiveSystemCapabilityUpdatedNotification object:nil];
-    [[NSNotificationCenter defaultCenter] addObserver:self selector:@selector(sdl_systemCapabilityResponseNotification:) name:SDLDidReceiveGetSystemCapabilitiesResponse object:nil];
-    [[NSNotificationCenter defaultCenter] addObserver:self selector:@selector(sdl_hmiStatusNotification:) name:SDLDidChangeHMIStatusNotification object:nil];
```

Comment:
I think you still want to be registered for `SDLDidChangeHMIStatusNotification`, otherwise the `currentHMILevel` is always `NONE`.

Reply:
Strange, did I forget to add the notification register? I added it now.